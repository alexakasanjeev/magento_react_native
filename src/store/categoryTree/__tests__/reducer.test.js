import categoryTreeReducer, { INITIAL_STATE } from '../reducer';
import { MAGENTO } from '../../../constants';

describe('category tree', () => {
  test('should return the initial state', () => {
    // Exercise
    const newState = categoryTreeReducer(undefined, {});

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle catgeory tree success', () => {
    // Setup
    const action = {
      type: MAGENTO.CATEGORY_TREE_SUCCESS,
      payload: {
        categoryTree: {
          children_data: [],
          id: 2,
          is_active: true,
          level: 1,
          name: 'Default Category',
          parent_id: 1,
          position: 1,
          product_count: 1181,
        }
      }
    };

    // Exercise
    const newState = categoryTreeReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle catgeory tree failure', () => {
    // Setup
    const action = {
      type: MAGENTO.CATEGORY_TREE_FAILURE,
      payload: {
        errorMessage: 'Unable to fetch category tree',
      }
    };

    // Exercise
    const newState = categoryTreeReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });
});
