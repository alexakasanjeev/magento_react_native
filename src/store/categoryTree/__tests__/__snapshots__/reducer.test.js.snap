// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`category tree should handle catgeory tree failure 1`] = `
Object {
  "errorMessage": "Unable to fetch category tree",
  "status": "ERROR",
}
`;

exports[`category tree should handle catgeory tree success 1`] = `
Object {
  "children_data": Array [],
  "errorMessage": "",
  "id": 2,
  "is_active": true,
  "level": 1,
  "name": "Default Category",
  "parent_id": 1,
  "position": 1,
  "product_count": 1181,
  "status": "SUCCESS",
}
`;

exports[`category tree should return the initial state 1`] = `
Object {
  "errorMessage": "",
  "status": "LOADING",
}
`;
